### ISSUE TITLE
I am getting this error when i try to download.

### ISSUE THREAD

**Issue by KashTag** [vn-ki/anime-downloader#5]
Command I had entered:
anime dl --episodes 78:102 --download-dir D:\Harsh http://kissanime.ru/Anime/Fairy-Tail-2014-Dub

![image](https://user-images.githubusercontent.com/39453935/41392112-49cd774c-6fbc-11e8-9600-dbcbc237632a.png)

**Comment by vn-ki** [comment]
I have released a bug fix version. Can you please try it out and report back?

```
pip install -U anime-downloader[cloudflare]
```

You have to install with cloudflare support if you want kissanime to be supported. If clouflare is not available, it will just search using that url.

I would suggest using 9anime, because it would be faster and is more tested. (Search is available using 9anime)

**Comment by KashTag** [comment]
Okay so this was weird. I uninstalled the normal one with this cloudfare supported one. After installing it gave a weird message saying anime-downloader 2.4.1 didn't support cloudfare even though i had pasted the exact same code. Nevertheless I tried downloading through 9anime but it gave another error saying 'No connection adapters were found for [URL]'. So I instead tried using KissAnime and well instead of using that link it showed me a bunch of different anime and movies(though all of them were related to the anime whose link i had originally tried to search..in this case Fairy Tail). I chose the one I wanted. Another thing is that all these options were definitely from 9anime( Kissanime calls it Fairy Tail 2014 while 9anime calls it Fairy Tail Series 2 or something and i got the latter as an option). And it worked. Still it's a bummer it didn't use KissAnime since 9anime(which I'm guessing uses playercdn or something) has like 1.5x bigger video file sizes for the exact same quality as compared to KissAnime(which uses RapidVideo). 
TL;DR - Not working when directly tried 9anime but when trying Kissanime it instead uses 9anime.
Sorry for the long read. Thanks for all the effort you've made trying to make this but the bigger file sizes of 9anime can be a real downer, especially when downloading whole anime seasons.

**Comment by vn-ki** [comment]
Okay. So yeah, this was indeed weird. Can you paste the traceback or the error you received when trying to download directly from 9anime? Also use `-ll DEBUG` argument while you are at it.

BTW, this tool downloads uses rapidvideo servers for both 9anime and kissanime.

Also I know this is too much to ask, but can you run the following commands and paste their results?
```
anime dl http://kissanime.ru/Anime/Kochin-Pa --url -ll DEBUG
```
```
anime dl https://www5.9anime.is/watch/steinsgate.1rx --url -ll DEBUG
```

**Comment by KashTag** [comment]
![image](https://user-images.githubusercontent.com/39453935/41403419-7392cdfe-6fe2-11e8-8f74-b225bc8f68d5.png)
 
This is what I get from trying to download directly via 9anime. Sorry I couldn't properly use -11 DEBUG as you can see.

 
![image](https://user-images.githubusercontent.com/39453935/41403522-b2f7befa-6fe2-11e8-83a5-2b2478002b26.png)
![image](https://user-images.githubusercontent.com/39453935/41403545-bec43196-6fe2-11e8-9e8d-5f5b319e23de.png)

This is after trying out the KochinPa command. I entered 3 when it asked me to choose since that looked the most similar.

![image](https://user-images.githubusercontent.com/39453935/41403651-03c1e360-6fe3-11e8-9c44-61165a5c5694.png)
![image](https://user-images.githubusercontent.com/39453935/41403673-111ee8aa-6fe3-11e8-8d97-ab15dea22eee.png)

After that it started downloading the URLs properly

![image](https://user-images.githubusercontent.com/39453935/41403811-6d1726b8-6fe3-11e8-8411-40e70583cfdc.png)
![image](https://user-images.githubusercontent.com/39453935/41403823-768bba88-6fe3-11e8-91ca-cde101ed3b7d.png)

Hmm it worked properly this time even with 9anime. Could I know what you did differently that it worked now while it didn't for me?

**Comment by KashTag** [comment]
From what I can see the two things different were that your 9anime link doesnt have the weird number that changes for every episode which it usually does since 9anime directly opens up an episode instead of some kind of anime page. Also your link is a ww5.9anime while i was using ww6.9anime.

**Comment by vn-ki** [comment]
Hmm. Seems like 9anime worked. I didn't change anything. Perhaps you could share what url you were trying?

As for kissanime, can you do, `python -c "import cfscrape"` and paste the output. If I had to guess, I think it is because you don't have node js installed, which is a dependency for cfscrape.

As for `-ll`, you have to use the alphabet `l` not the number `1` (I know some monospace fonts can be confusing :P).

**Comment by vn-ki** [comment]
> From what I can see the two things different were that your 9anime link doesnt have the weird number that changes for every episode which it usually does since 9anime directly opens up an episode instead of some kind of anime page. Also your link is a ww5.9anime while i was using ww6.9anime.

It shouldn't matter, if I remember correctly. If you could share the url, I can try it on my machine.

**Comment by KashTag** [comment]
![image](https://user-images.githubusercontent.com/39453935/41404553-244715fe-6fe5-11e8-8568-ef3e013c4991.png)

Also its working now, that my w6.9anime link i.e. Guess I was making some noobish mistake. Sorry :P

Though Kissanime links still redirect to 9anime, but ive realized the file sizes are only small for Fairy Tail, and even then the quality is drastically low as compared to 9anime. Sorry again for arguing about that. 9anime is good as any other.

**Comment by vn-ki** [comment]
Ah. Good to know it's working. :)

I'll close this issue now. If you run into anything else, feel free to open an issue.

**Comment by vn-ki** [comment]
Also, if you still want kissanime, you can do `pip install cfscrape` and make sure you have [node-js](https://nodejs.org/en/) installed.

**Comment by KashTag** [comment]
Thanks for that. I'll keep that in mind. And I'm so sorry again for all this ruckus. In the end it was probably some noobish mistake i made. Maybe your updated 2.4.1 version fixed it but i doubt that. Thanks a lot for bearing with me man.
